### ISSUE TITLE
make just a select(count()) query takes a huge amount of time compered to select(count(),map())

### ISSUE THREAD

**Issue by CanaliDavide** [pola-rs/polars#8329]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Issue description

I have a dataset this csv dataset ([https://www.kaggle.com/datasets/kumarajarshi/life-expectancy-who?resource=download](dataset)) concatenated n times so that I have 1GB, 3GB and 6GB files.

Then I was making some tests and I found out that if I run this code:

```
fn main() -> Result<()>{

    let args: Vec<String> = args().collect();
    let df = LazyCsvReader::new(&args[1]).with_rechunk(true).finish().unwrap();

    df.select([
        count(),
        col("Adult Mortality").map(|s| {Ok(Some(s / 100))}, GetOutput::from_type(DataType::UInt16)).sum()
        ])
        .collect()?;

    Ok(())
}
```
with hyperfine --warmup 3 run code -- data_3GB.csv it takes around 4.5 seconds with both rechunk True and False.

Instead if I run this code:
```
fn main() -> Result<()>{
    let args: Vec<String> = args().collect();

    let df = LazyCsvReader::new(&args[1]).with_rechunk(true).finish().unwrap();
    
    df.select([
        count()
    ]).collect()?;

    Ok(())
}
```
hyperfine --warmup 3 run code -- data_3GB.csv it takes around 40 seconds with rechunk True and 18 seconds with rechunk False.

Note:
Both count() return the same value.
The same issue happens with the 6GB file.

### Reproducible example

Download the dataset from [https://www.kaggle.com/datasets/kumarajarshi/life-expectancy-who?resource=download](dataset) then fill all the missing values with a value of choice (I used "5").

Create the 3 files with a bash code (linux Ubuntu):
```
n=3030

>data_tail.csv
>data_1_tail.csv
>data_3_tail.csv
>data_6_tail.csv

cat data.csv > data_1.csv
cat data.csv > data_3.csv
cat data.csv > data_6.csv

tail -n +2 data.csv > data_tail.csv

# Concatenate the input file with itself multiple times
for (( i=1; i<=$n; i++ ))
do
  cat data_tail.csv >> data_1_tail.csv
done

for (( i=1; i<=3; i++ ))
do
  cat data_1_tail.csv >> data_3_tail.csv
done

for (( i=1; i<=2; i++ ))
do
  cat data_3_tail.csv >> data_6_tail.csv
done

cat data_1_tail.csv >> data_1.csv
cat data_3_tail.csv >> data_3.csv
cat data_6_tail.csv >> data_6.csv

```

run examples codes with hyperfine:
with hyperfine --warmup 3 run code -- data_3.csv

### Expected behavior

I would expect the latter to run at the same speed as the former, if not faster.

### Installed versions

<details>
polars = {version = "0.28.0", features = ["lazy", "strings" , "csv-file", "lazy_regex", "concat_str", "performant"]}
</details>

**Comment by ritchie46** [comment]
What is the output if you set `POLARS_VERBOSE=1`?

**Comment by ritchie46** [comment]
Right, I think the one `map` leads to a projection pushdown of selecting a single columns, where the count doesn't trigger projection pushdown so all columns are loaded.

A reminder that we should add the `count` special case query. I will see if I can get to this this week.

**Comment by coastalwhite** [comment]
This is done now. Closing.
